Record time taken to complete review

rachita19082 opened the pull request:
This pull request resolves issue #16

rachita19082 committed:
Added functionality to record time taken to complete review

rachita19082 commented:
@lacabra I have made some changes to the code which I think will resolve the issue. However, I was unable to figure out a way to test the code. Please let me know if there is any way for me to test the code and if any changes are required. While committing, I also got the error _"husky > pre-commit hook failed (add --no-verify to bypass)"_, can you let me know what could be the reason behind this?

lacabra commented:
@rachita19082: I have expanded the README documentation in 48ba60a, refer to the [Development Environment](https://github.com/lacabra/validate-digitalpublicgoods#-development-environment) section.

If you follow the updated instructions, your local environment will open PRs upon submit of the form to your clone of the publicgoods-candidates repos (instead of the UNICEF one), and that is the way how you can test whether your changes work as intended. Kindly link a successful PR to your clone to show that the functionality works as intended.

lacabra commented:
@rachita19082: the precommit husky check validates the linting of your files. You should see the actual error by running:
```bash
npm run link
```
which you can fix manually, or automatically by running:
```
npm run link:fix
```

You can typically find this information in `package.json` ;)

rachita19082 committed:
Fixed linting issue

rachita19082 committed:
Updated functionality to record time taken

rachita19082 committed:
Updated code

rachita19082 commented:
@lacabra I have fixed the linting issue and corrected a few mistakes in my code. However, I am facing problems with the OAuth app. What needs to be added to the Homepage url and authorization callback URL? Currently, I have added http://localhost:3000 as the Homepage url and http://localhost:3000/callback as the callback URL but it doesn't seem to work.

rachita19082 commented:
@lacabra I have tested the functionality of the code in this [PR](https://github.com/rachita19082/publicgoods-candidates-test/pull/1/commits/951c3e0c322b5efda09f4a20facd3540e3790299). The time is shown in seconds. Please let me know if any other changes are needed.

lacabra committed:
Merge branch 'main' into main

lacabra commented on pages/projects.js:
```diff
@@ -80,7 +82,7 @@ export default function Projects() {
   } else {
     if (currentReview) {
-      return <Review submission={currentReview} />;
+      return <Review submission={currentReview} start_time={startTime} />;
```
why do you use `start_time` here? for consistency, I'd rather use `startTime={startTime}` which should be perfectly fine. Curious to hear your rationale for switching between one and the other?

lacabra commented on pages/review.js:
```diff
@@ -391,7 +391,11 @@ export default function Review(props) {
+          <OpenPR
+            answer={result}
+            projectName={props.submission.name}
+            startTime={props.start_time}
```
If you change the one above, then again here you would use `startTime={props.startTime}`. Given that all the codebase uses camelCase, I see no reason to break the consistency here :)

rachita19082 replied:
@lacabra I try to avoid using the same name for the parameter and the state because I tend to get confused about what is being referenced and sometimes it has led to errors as well. But I agree that here it is better to keep the names consistent. I have made the required changes in commit a657fd5.

lacabra commented:
@rachita19082, the linter was added to the CI recently, and it now fails. Kindly take a look, and make sure that the `CI/build` passes before this pull request can be merged. Thank you

rachita19082 committed:
Changed parameter name

rachita19082 committed:
Merge branch 'main' of https://github.com/rachita19082/validate-digitalpublicgoods into main

rachita19082 commented:
> @rachita19082, the linter was added to the CI recently, and it now fails. Kindly take a look, and make sure that the `CI/build` passes before this pull request can be merged. Thank you

I have fixed the linting issues but the build requires approval. Please approve and I will make the necessary changes if the build fails again.

lacabra committed:
Merge branch 'main' into main

lacabra reviewed (approved):
Thanks @rachita19082, everything looks good now! 👍
